mm2src/mm2_bitcoin/keys/src/segwitaddress.rs

```diff
@@ -221,4 +226,53 @@ mod tests {
         );
         assert_eq!(addr.address_type(), Some(AddressType::P2wsh));
     }
+
+    #[test]
+    // https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki#test-vectors
+    fn test_invalid_segwit_addresses() {
```

Comment:
it's covered here https://github.com/KomodoPlatform/atomicDEX-API/blob/d950555d2d78a42e40c3e59a4deb3963894dd91d/mm2src/mm2_main/src/mm2_tests.rs#L1688 We check the hrp against the hrp from coin config outside the `from_str` function since the hrp from config can't be passed to `from_str`
So this address `tc1qw508d6qejxtdg4y5r3zarvary0c5xw7kg3g4ty` from the test vectors when converted to `SegwitAddress` using `from_str` will be valid

Reply:
This where the comparison happens https://github.com/KomodoPlatform/atomicDEX-API/blob/d950555d2d78a42e40c3e59a4deb3963894dd91d/mm2src/coins/utxo/utxo_common.rs#L3815-L3822

Reply:
Great, thought you forgot it